packages/transaction-controller/src/TransactionController.ts

```diff
@@ -338,16 +338,19 @@ export class TransactionController extends BaseController<
    * @param opts - Additional options to control how the transaction is added.
    * @param opts.deviceConfirmedOn - An enum to indicate what device confirmed the transaction.
    * @param opts.origin - The origin of the transaction request, such as a dApp hostname.
+   * @param opts.requireApproval - Whether to skip creating approval for transactions.
    * @returns Object containing a promise resolving to the transaction hash if approved.
    */
   async addTransaction(
     transaction: Transaction,
     {
       deviceConfirmedOn,
       origin,
+      requireApproval = true,
```

Comment:
Rather than default here, can we leave it without a default and pass it to `processApproval` as possibly undefined and then have a single explicit `requireApproval !== false` before requesting the approval to ensure the only way to disable is with an explicit `false`.

Currently you could pass anything that is falsy like `null` or `0` and it would disable the approval.

Reply:
Thats a good spot Matthew, committed the suggested way.